akka-stream/src/main/boilerplate/akka/stream/scaladsl/ZipLatestWithApply.scala.template

```diff
@@ -57,50 +59,55 @@ class ZipLatestWith1[[#A1#], O] (val zipper: ([#A1#]) ⇒ O) extends GraphStage[
         }
       )
 
-      [#setHandler(
-        in0,
-        new InHandler {
-          override def onPush() = {
-            value0 = grab(in0)
-            hasValue0 = true
-            if (waitingForTuple && hasAllValues) {
-                pushTuple()
-                waitingForTuple = false
-                pullAllIfNeeded()
-            }
-          }
-
-          override def onUpstreamFinish(): Unit = {
-            if (!isAvailable(in0)) completeStage()
-            willShutDown = true
-          }
-        }
-      )#
+      [#setHandler(in0, inlet0)#
       ]
 
-      private def hasAllValues = [#hasValue0#&&]
+      private def hasAllValues = [#inlet0.hasValue#&&]
 
-      private def pushTuple(): Unit = {
-        push(out, zipper([#value0#,]))
+      private def pushOutput(output: O): Unit = {
+        push(out, output)
         if (willShutDown) completeStage()
-        [#last0 = value0#
-        ]
-        hasLast = true
+        lastOutput = output
+        hasLastOutput = true
       }
 
-      private def pushIfFirstTupleOrDifferentTuple() =
-        if (hasLast) {
-          if ([#value0 != last0#||]) pushTuple() else waitingForTuple = true
+      private def pushIfFirstTupleOrDifferentOutput() = {
+        val output = zipCurrentTuple
+        if (hasLastOutput) {
+          if (lastOutput != output) pushOutput(output) else waitingForTuple = true
```

Comment:
That's still equality check though? For instance equality it would have to be `lastOutput.ne(output)`.

Reply:
And I wonder if we _really_ need to keep track of the previous tuple at all, if we don't do said equality based deduplication.

Reply:
see my other inline comment on the corresponding test: this check is for the case when we pull again on the source and nothing happened upstream: we don't want to re-emit again